File: mambo-shared/mambo-protocol/mambo-protocol-client/src/main/java/org/mambo/protocol/client/messages/CharacterSelectedForceMessage.java


// Generated on 01/04/2013 14:54:21
package org.mambo.protocol.client.messages;

import java.util.*;
import org.mambo.protocol.client.types.*;
import org.mambo.protocol.client.enums.*;
import org.mambo.protocol.client.*;
import org.mambo.core.io.*;

public class CharacterSelectedForceMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6068;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int id;
    
    public CharacterSelectedForceMessage() { }
    
    public CharacterSelectedForceMessage(int id) {
        this.id = id;
    }
    
    @Override
    public void serialize(DataWriterInterface writer) {
        writer.writeInt(id);
    }
    
    @Override
    public void deserialize(DataReaderInterface reader) {
        id = reader.readInt();
        if (id < 1 || id > 2147483647)
            throw new RuntimeException("Forbidden value on id = " + id + ", it doesn't respect the following condition : id < 1 || id > 2147483647");
    }
    
}

File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/character/choice/CharacterSelectedForceMessage.scala





















// Generated on 05/03/2013 10:34:36
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait CharacterSelectedForceMessage extends MessageDeserializer {
  val messageId = 6068

  protected trait Message extends BaseMessage with Serializable {
    val messageId = CharacterSelectedForceMessage.messageId

    val id: Int

    def serialize(buf: Buffer) {
      buf.writeInt(id)
    }

  }

  protected def create(id: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val id = buf.readInt()

    create(id)
  }

}

object CharacterSelectedForceMessage extends CharacterSelectedForceMessage {
  case class MessageImpl(id: Int) extends Message
  type Target = MessageImpl

  override def create(id: Int) = MessageImpl(id)

  def apply(id: Int) = create(id)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

